### ISSUE TITLE
Client transitions don't handle data request errors well

### ISSUE THREAD

**Issue by nampas** [redfin/react-server#189]
Repro:
1) Go to Page A
2) Click on a link to execute a client transition to Page B. Page B should make a data request that 500's
3) Attempt a client transition back to Page A

Result:
Get the following console error, and navigation fails:
`error: [react-server.core.ClientController] onNavigate error Object {status: 404, message: "No Route!"}`

Expected Result:
Transitioning away from a page that had a data request error should succeed

Note that this repros when `bundleData` is on or off

**Comment by gigabo** [comment]
Whoa... that's not good.  Thanks for reporting @nampas.

**Comment by gigabo** [comment]
Hi @nampas.  I'm having trouble reproducing this.

Please have a look at #197 and see if anything jumps out at you.

**Comment by nampas** [comment]
Hrm ok thanks for throwing that example together, I'll take a look and get back to you.

**Comment by nampas** [comment]
Ok, so it turns out that we had a RootElement that didn't resolve in the case of a data request failing. That made react-server think that page B never finished rendering (`Navigator.finishRoute` was never invoked). Then, when we tried to execute the transition back to page A, `Navigator.startRoute` wouldn't emit `navigateStart` because it was waiting for the current page load to finish.

Closing this ticket, but it might be worth adding a timeout for finishRoute()
